.github/workflows/build.yml

```diff
@@ -0,0 +1,47 @@
+name: Check and build
+
+on:
+  pull_request:
+  push:
```

Comment:
Please use `pull_request_target` instead of `pull_request`, it's safer. And remove `push` trigger (no need for them, what do you think?)